### PR TITLE
Generate contracts nt

### PR DESCRIPTION
**Pull request by nolantait**

### CODE REVIEW

File: app/controllers/batches_controller.rb [hunk 1]
```diff
@@ -40,7 +44,7 @@ def create
   # PATCH/PUT /batches/1
   def update
     if @batch.update(batch_params)
-      redirect_to @guild, notice: 'Batch was successfully updated.'
+      redirect_to batches_path, notice: 'Batch was successfully updated.'
```
Comment (SENC275) [inline app/controllers/batches_controller.rb]: Use  locale  for 'Batch was successfully updated.' ?

File: app/controllers/contracts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      redirect_to root_path, notice: t('contracts.create.failure')
+    end
+  end
+
+  # PATCH/PUT /contracts/1
+  def update
+    if @contract.generate_output(datum_params)
+      redirect_to root_path, notice: t('contracts.update.success')
+    else
+      redirect_to @contract, notice: 'Could not save your answer.'
```
Comment (SENC275) [inline app/controllers/contracts_controller.rb]: use locale?

File: spec/factories/contract_data.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+FactoryBot.define do
+  factory :contract_datum do
+    contract nil
```
Comment (SENC275) [inline spec/factories/contract_data.rb]: user factory: :contract & factory: :datum

Comment (nolantait) [reply spec/factories/contract_data.rb]: Fixed in a later commit.

File: app/controllers/contracts_controller.rb (hunk 2 above)
Comment (nolantait) [reply app/controllers/contracts_controller.rb]: Will fix this all in latest branch

**Pull request merged**